packages/terra-collapse/tests/jest/Collapse.test.jsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import IconCaretRight from 'terra-icon/lib/icon/IconCaretRight';
+import Collapse from '../../src/Collapse';
+
+
+describe('Collapse', () => {
+  // Snapshot Tests
+  it('should render a default collapse', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse).toMatchSnapshot();
+  });
+
+  it('should render an animated collapse', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" isAnimated>Test</Collapse>);
+    expect(collapse).toMatchSnapshot();
+  });
+
+  it('should render an initially open collapse', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" isInitiallyOpen>Test</Collapse>);
+    expect(collapse).toMatchSnapshot();
+  });
+
+  it('should mount a collapse', () => {
+    const collapse = mount(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse).toMatchSnapshot();
+  });
+
+  it('should render a collapse', () => {
+    const collapse = render(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse).toMatchSnapshot();
+  });
+
+  // Event Tests
+  it('should render an open collapse when clicked', () => {
+    const collapse = mount(<Collapse closedButtonText="Show">Test</Collapse>);
+    collapse.find('button').simulate('click');
+    expect(collapse).toMatchSnapshot();
+  });
+
+  // Prop Tests
+  it('should set buttonAttrs prop correctly', () => {
+    const collapse = shallow(<Collapse buttonAttrs={{ variant: 'link' }} closedButtonText="Show">Test</Collapse>);
+    expect(collapse.unrendered.props.buttonAttrs).toEqual({ variant: 'link' });
+  });
+
+  it('should set children prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse.unrendered.props.children).toEqual('Test');
+  });
+
+  it('should set closedButtonText prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse.unrendered.props.closedButtonText).toEqual('Show');
+  });
+
+  it('should set icon prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse).toMatchSnapshot();
```

Comment:
The icon is being set with the default prop so its not explicitly set here and then I'm using snapshots to verify the icon renders correctly and capturing that in a snapshot.